internal/controllers/user_integrations_controller.go

```diff
@@ -1933,6 +1953,49 @@ func (udc *UserDevicesController) registerDeviceTesla(c *fiber.Ctx, logger *zero
 	return c.SendStatus(fiber.StatusNoContent)
 }
 
+func (udc *UserDevicesController) getTeslaAuthFromCache(ctx context.Context, userID string) (*RegisterDeviceIntegrationRequest, error) {
+	user, err := udc.usersClient.GetUser(ctx, &pb.GetUserRequest{Id: userID})
+	if err != nil {
+		return nil, errors.Wrap(err, "could not fetch user information")
+	}
+
+	if user.EthereumAddress == nil {
+		return nil, fmt.Errorf("missing wallet address for user")
+	}
+
+	cacheKey := fmt.Sprintf(TeslaFleetAuthCacheKey, *user.EthereumAddress)
+	encTeslaAuth, err := udc.redisCache.Get(ctx, cacheKey).Result()
+	if err != nil {
+		udc.log.Err(err).Str("Cache Key", cacheKey).Msg("Error occurred retrieving tesla auth from cache using the key")
+		return nil, errors.Wrap(err, "could not retrieve Tesla credentials")
+	}
+	if len(encTeslaAuth) == 0 {
+		return nil, fmt.Errorf("no credential found")
+	}
+	decrypted, err := udc.cipher.Decrypt(encTeslaAuth)
+	if err != nil {
+		udc.log.Err(err).Str("cacheKey", cacheKey).Msg("failed to decrypt tesla auth from cache using key")
+		return nil, errors.Wrap(err, "failed to decrypt tesla token")
```

Comment:
Nit: often it's helpful to either log or return an error but not do both. The reasoning for this is if someone uses this function outside of its current use case they will be forced to log errors even if they do not care about them. Or they will not have meaningful error messages because they were suppressed. I believe we are doing this because we do not want to provide verbose messages in the HTTP response, but it should be up to the function that is handling the response, what gets sent back and what gets logged not the function it is calling.

Reply:
Normally I would agree, but my intention for this is to decongest the main controller function and not for external use. That being said, there is no reason why a calling function would not want to log errors on the server to help with debugging this further. Also the error isn't being suppressed, if you look at the function its still being returned.

Reply:
I think the ask would be to log as high up as possible, something like line 1825.

Reply:
I think this was done, yeah?